File: blok2/jpa/src/main/java/nl/bramjanssens/util/EntityManagerProducer.java
package nl.bramjanssens.util;

import jakarta.enterprise.inject.Produces;
import jakarta.inject.Singleton;
import jakarta.persistence.EntityManager;
import jakarta.persistence.Persistence;

@Singleton
public class EntityManagerProducer {

    @Produces
    public EntityManager em() {
        return Persistence
                .createEntityManagerFactory("MySQL").createEntityManager();
    }
}

File: blok2/jpa/src/main/java/nl/bramjanssens/util/EntityManagerProducer.kt
package nl.bramjanssens.util

import jakarta.inject.Singleton
import jakarta.persistence.Persistence

// Kotlin :-)
@Singleton
class EntityManagerProducer {

    companion object MySQL {
        private val entityManagerFactory = Persistence.createEntityManagerFactory("MySQL")

        fun connection(): jakarta.persistence.EntityManagerFactory = entityManagerFactory
    }

    // Java:
    // @Produces
    // public EntityManager emf() {
    //     return Persistence.createEntityManagerFactory("MySQL");
    // }
}
